### PR TITLE
Feature/57 remove need for recursive queries

### PR DESCRIPTION
**Pull request by cryptomental**
This PR solves #57.

I also updated copyright messages and updated Oraclize test case to test also for UpdatedLastPrice event.

### CODE REVIEW

**Comment by pelsasser** [comment]
@cryptomental - this all looks great - I want to pull the changes and test it a bit locally and then will merge.  I assume this reduces deployment costs a bit and also the needed gas for our `_callback` since we are scheduling another query during the callback

File: contracts/oraclize/MarketContractOraclize.sol [hunk 1]
```diff
@@ -134,18 +111,20 @@ contract MarketContractOraclize is MarketContract, usingOraclize {
...
+        // Require that sufficient funds are available to pay for the query.
+        require(oraclize_getPrice(ORACLE_DATA_SOURCE, QUERY_CALLBACK_GAS) < this.balance);
+        // Require expiration time in the future.
+        require(EXPIRATION > now);
+        // Future timestamp must be within 60 days from now.
+        // https://docs.oraclize.it/#ethereum-quick-start-schedule-a-query-in-the-future
+        require(EXPIRATION - now <= 60 * 60 * 24 * 60);
+
+        bytes32 queryId = oraclize_query(
+            EXPIRATION - now,
```
Comment (pelsasser) [inline contracts/oraclize/MarketContractOraclize.sol]: I think we can use an absolute (not relative) time stamp here. 

https://docs.oraclize.it/#ethereum-quick-start-schedule-a-query-in-the-future

It sounds like using `EXPIRATION` only should work - maybe you can test and clarify?

File: contracts/oraclize/MarketContractOraclize.sol [hunk 2]
```diff
@@ -134,18 +111,20 @@ contract MarketContractOraclize is MarketContract, usingOraclize {
...
-            );
-            validScheduledQueryIDs[queryId] = true;
-        }
+        // Require that sufficient funds are available to pay for the query.
+        require(oraclize_getPrice(ORACLE_DATA_SOURCE, QUERY_CALLBACK_GAS) < this.balance);
+        // Require expiration time in the future.
+        require(EXPIRATION > now);
+        // Future timestamp must be within 60 days from now.
+        // https://docs.oraclize.it/#ethereum-quick-start-schedule-a-query-in-the-future
+        require(EXPIRATION - now <= 60 * 60 * 24 * 60);
```
Comment (pelsasser) [inline contracts/oraclize/MarketContractOraclize.sol]: Can we move this to early on in the constructor perhaps?  I think that may make more sense than here.

File: contracts/oraclize/MarketContractOraclize.sol [hunk 3]
```diff
@@ -30,19 +29,12 @@ contract MarketContractOraclize is MarketContract, usingOraclize {
     // constants
     string public ORACLE_DATA_SOURCE;
     string public ORACLE_QUERY;
-    uint public ORACLE_QUERY_REPEAT;
     uint constant public QUERY_CALLBACK_GAS = 180000;
```
Comment (pelsasser) [inline contracts/oraclize/MarketContractOraclize.sol]: I assume we can lower this now that the code path is much simpler.

File: contracts/oraclize/MarketContractOraclize.sol (hunk 2 above)
Comment (pelsasser) [inline contracts/oraclize/MarketContractOraclize.sol]: thoughts on using a constant variable for the 60 days value converted to seconds?

File: contracts/oraclize/MarketContractOraclize.sol (hunk 1 above)
Comment (cryptomental) [reply contracts/oraclize/MarketContractOraclize.sol]: All right, I will test this asap.

File: contracts/oraclize/MarketContractOraclize.sol (hunk 2 above)
Comment (cryptomental) [reply contracts/oraclize/MarketContractOraclize.sol]: Will be moved to constructor.

File: contracts/oraclize/MarketContractOraclize.sol (hunk 3 above)
Comment (cryptomental) [reply contracts/oraclize/MarketContractOraclize.sol]: Yes, we can lower it, tests say ~115k, coverage network a bit more, so 130000 - 140000 should be a safe value.

File: contracts/oraclize/MarketContractOraclize.sol (hunk 2 above)
Comment (cryptomental) [reply contracts/oraclize/MarketContractOraclize.sol]: Fine. I will create a variable for that.

**Comment by cryptomental** [comment]
@pelsasser I added a commit that reworks the PR after your findings:

* use absolute timestamp to oraclize_query
* moved time checks to constructor
* introduced a constant variable SECONDS_PER_SIXTY_DAYS
* lowered Oraclize callback cost to 140000.
 
I hope this can be merged now.

**Comment by cryptomental** [comment]
I am still experimenting a bit with oraclize callback gas.

**Comment by cryptomental** [comment]
Ok, all checks passed, we can merge!

**Pull request merged**